### PR TITLE
Create Journal Entry Form with Styles and Functionality

### PR DESCRIPTION
**Pull request by ajaguasin**
- Create the journalEntry Screen and journal entry form component
- When the Journal Screen is finished, we will connect JournalEntry Screen and add navigation and the proper props, realm/context methods

### CODE REVIEW

File: js/components/JournalEntryForm/JournalEntryForm.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      imageSource: []
+    };
+  }
+
+  removeImg = img => {
+    console.log(this.state.imageSource);
+    console.log(img);
```
Comment (siddharthparmar7) [inline js/components/JournalEntryForm/JournalEntryForm.js]: remove this console log

File: js/components/JournalEntryForm/JournalEntryForm.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    const oldArr = this.state.imageSource;
+    const filteredArr = oldArr.filter(element => {
+      return element.uri !== img.uri;
+    });
+    this.setState({ imageSource: filteredArr });
+    console.log(this.state.imageSource);
```
Comment (siddharthparmar7) [inline js/components/JournalEntryForm/JournalEntryForm.js]: remove this console log

File: js/components/JournalEntryForm/JournalEntryForm.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    const options = {
+      title: "Select Photo",
+      storageOptions: {
+        skipBackup: true,
+        path: "images"
+      }
+    };
+
+    ImagePicker.showImagePicker(options, response => {
+      console.log("Response = ", response);
```
Comment (siddharthparmar7) [inline js/components/JournalEntryForm/JournalEntryForm.js]: remove all the console logs.

File: js/components/JournalEntryForm/JournalEntryForm.js [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        console.log("User cancelled image picker");
+      } else if (response.error) {
+        console.log("ImagePicker Error: ", response.error);
+      } else if (response.customButton) {
+        console.log("User tapped custom button: ", response.customButton);
+      } else {
+        const source = { uri: response.uri };
+
+        // You can also display the image using data:
+        // const source = { uri: 'data:image/jpeg;base64,' + response.data };
```
Comment (siddharthparmar7) [inline js/components/JournalEntryForm/JournalEntryForm.js]: this comments from docs are for your reference. Please remove them or use them.

File: js/components/JournalEntryForm/JournalEntryForm.js [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+      body: JSON.stringify(body.content)
+    });
+    console.log(res);
+  };
+
+  render() {
+    return (
+      <Form
+        onSubmit={values => {
+          console.log(JSON.stringify(values, 0, 2));
```
Comment (siddharthparmar7) [inline js/components/JournalEntryForm/JournalEntryForm.js]: remove this console log

File: js/components/JournalEntryForm/JournalEntryForm.js [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+    const token =
+      "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwczpcL1wvYXBpLnNwZWFrYm94LXN0YWdpbmcuY2FcL2FwaVwvYXV0aFwvbG9naW4iLCJpYXQiOjE1NDUwOTMzMTAsImV4cCI6MTU0NTA5NjkxMCwibmJmIjoxNTQ1MDkzMzEwLCJqdGkiOiJsNHg2YVg1aVNZb2xpWkJhIiwic3ViIjoxMiwicHJ2IjoiODdlMGFmMWVmOWZkMTU4MTJmZGVjOTcxNTNhMTRlMGIwNDc1NDZhYSJ9.YnZtVv-zTKZBlGL1OqSn6m4js04ppiptSxiiMXXTrvE";
+    const res = await apiPromise("/journal-entries", {
+      method: "POST",
+      headers: {
+        Authorization: `Bearer${[token]}`
+      },
+      body: JSON.stringify(body.content)
+    });
+    console.log(res);
```
Comment (siddharthparmar7) [inline js/components/JournalEntryForm/JournalEntryForm.js]: remove console log

File: js/components/JournalEntryForm/JournalEntryForm.js [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+          imageSource: [...this.state.imageSource, source]
+        });
+      }
+    });
+  };
+
+  createJournalEntry = async body => {
+    console.log(body.content);
+    const token =
+      "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwczpcL1wvYXBpLnNwZWFrYm94LXN0YWdpbmcuY2FcL2FwaVwvYXV0aFwvbG9naW4iLCJpYXQiOjE1NDUwOTMzMTAsImV4cCI6MTU0NTA5NjkxMCwibmJmIjoxNTQ1MDkzMzEwLCJqdGkiOiJsNHg2YVg1aVNZb2xpWkJhIiwic3ViIjoxMiwicHJ2IjoiODdlMGFmMWVmOWZkMTU4MTJmZGVjOTcxNTNhMTRlMGIwNDc1NDZhYSJ9.YnZtVv-zTKZBlGL1OqSn6m4js04ppiptSxiiMXXTrvE";
```
Comment (siddharthparmar7) [inline js/components/JournalEntryForm/JournalEntryForm.js:47]: it's time, start using the realm for your token.

File: js/components/JournalEntryForm/JournalEntryForm.js [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+                  handleSubmit();
+                }}
+                style={styles.submitButton}
+              >
+                <Text style={styles.submitText}>submit</Text>
+              </TouchableOpacity>
+            </LinearGradient>
+            <TouchableOpacity
+              onPress={() => {
+                console.log("submit");
```
Comment (siddharthparmar7) [inline js/components/JournalEntryForm/JournalEntryForm.js]: you should be calling your on submit method not console logging a message.

File: js/components/JournalEntryForm/styles.js [hunk 9]
```diff
@@ -0,0 +1,85 @@
+import { StyleSheet, Dimensions } from "react-native";
+import globalStyles from "../../config/styles";
```
Comment (siddharthparmar7) [inline js/components/JournalEntryForm/styles.js]: remove unnecessary imports.

File: js/screens/JournalEntry/JournalEntry.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { View, Text } from "react-native";
+import JournalEntryForm from "../../components/JournalEntryForm";
+import LinearGradient from "react-native-linear-gradient";
+import moment from "moment";
+import styles from "./styles";
+import globalStyles from "../../config/styles";
+const JournalEntry = props => {
```
Comment (siddharthparmar7) [inline js/screens/JournalEntry/JournalEntry.js:9]: if you are not using props just do an empty function.

File: js/screens/JournalEntry/JournalEntry.js [hunk 11]
```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import PropTypes from "prop-types";
```
Comment (siddharthparmar7) [inline js/screens/JournalEntry/JournalEntry.js]: remove unnecessary improts

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
Remove all the console logs.
Start using Realm as your database to save user token.
Please make all the required changes. Thank you.

**Review by siddharthparmar7 — COMMENTED** [review]
Also, your form design does not match our designer's vision. Please style it well.

File: js/components/JournalEntryForm/JournalEntryForm.js (hunk 8 above)
Comment (ajaguasin) [reply js/components/JournalEntryForm/JournalEntryForm.js]: I will use this onPress event to navigate to the Journal Screen. When I receive the navigation props from the person who is using my feature, I will then add the functionality. Right now the functionality that works is the uploading the journal to SpeakBox's api

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
There are still some console logs in your code. Please remove them.

File: js/App.js [hunk 12]
```diff
@@ -1,6 +1,7 @@
 import React, { Component } from "react";
 import AppNavigator from "./navigation/RootStackNavigator";
 import { UserProvider } from "./context/UserContext";
+import JournalEntry from "./screens/JournalEntry";
```
Comment (siddharthparmar7) [inline js/App.js:4]: remove this import

**Review by siddharthparmar7 — APPROVED** [review]

**Pull request merged**